apps/cbportal/base/cloudbroker__machine/methodclass/cloudbroker_machine.py

```diff
@@ -103,6 +104,35 @@ def start(self, machineId, reason, **kwargs):
             j.apps.cloudbroker.machine.untag(vmachine.id, "start")
         self.cb.actors.cloudapi.machines.start(machineId=machineId)
 
+    @auth(['level1', 'level2', 'level3'])
+    @wrap_remote
+    def restore(self, machineId, reason, **kwargs):
+        """
+        Restore a deleted machine
+
+        :param machineId: id of the machine
+        """
+        ctx = kwargs['ctx']
+        ctx.env['JS_AUDIT'] = True
+        ctx.env['tags'] += " machineId:{}".format(machineId)
+        machine = self.models.vmachine.get(int(machineId))
```

Comment:
this is not safe call as machineId might not exist afaik this class has a safe way to get the object or raise NotFound exception